### PR TITLE
Hardware Key Support docs - additional troubleshooting info

### PR DESCRIPTION
**Pull request by Joerger**
Update hardware key support docs to cover re-login flow from role setting and add additional troubleshooting info.

This update is based on @stevenGravy's experience trying Hardware Key Support.

Updates https://github.com/gravitational/teleport/issues/20384

### CODE REVIEW

File: docs/pages/access-controls/guides/hardware-key-support.mdx
Comment (stevenGravy) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:113]: We only support Enterprise and Cloud so this can be removed.  It won't be visible

Comment (Joerger) [reply docs/pages/access-controls/guides/hardware-key-support.mdx:113]: Better to show it than have blanks in the page, this was an issue in the past - https://github.com/gravitational/teleport/issues/24211

**Review by lsgunn-teleport — COMMENTED** [review]
I made some wording suggestions in this PR.
https://github.com/gravitational/teleport/pull/29157
There were a couple of typos (like "have have" and "in order to perform safely authenticate you given your authentication settings"). I might also suggest using the more generic hardware security key instead of mentioning Yubikey explicitly (except where it's necessary).

**Review by lsgunn-teleport — APPROVED** [review]
LGTM 😄.

**Review by stevenGravy — APPROVED** [review]

File: docs/pages/access-controls/guides/hardware-key-support.mdx
Comment (zmb3) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:27]:
```suggestion
With touch required, hardware key support provides better security than per-session MFA.
```

File: docs/pages/access-controls/guides/hardware-key-support.mdx
Comment (zmb3) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:12]: It's not really the "session" that's stored on the filesystem, but the certificates. Should we be specific?

File: docs/pages/access-controls/guides/hardware-key-support.mdx
Comment (Joerger) [reply docs/pages/access-controls/guides/hardware-key-support.mdx:55]: I'll remove the examples as they are listed above

File: docs/pages/access-controls/guides/hardware-key-support.mdx
Comment (Joerger) [reply docs/pages/access-controls/guides/hardware-key-support.mdx:136]: I can update this in a follow up PR.

File: docs/pages/access-controls/guides/hardware-key-support.mdx
Comment (Joerger) [reply docs/pages/access-controls/guides/hardware-key-support.mdx:211]: I'll referece `tctl edit` as an example.

**Comment by alexfornuto** [comment]
@Joerger looks like this is still waiting on approval from members of two no-docs groups. This would be a good candidate to request for review in #docs.

File: docs/pages/access-controls/guides/hardware-key-support.mdx
Comment (ptgott) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:13]: Why is "User" capitalized here?

File: docs/pages/access-controls/guides/hardware-key-support.mdx
Comment (ptgott) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:17]:
```suggestion
when users start new sessions with Teleport services, such as the SSH Service, Kubernetes Service, Database Service, and so on. 
```
So we have consistent architectural terms throughout the docs

**Review by ptgott — APPROVED** [review]

**Review by AntonAM — APPROVED** [review]

**Review by GavinFrazar — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**
